### DISCUSSION TITLE
Rest API pagination helper

### DISCUSSION THREAD

**Discussion by hillairet** [SatelCreative/spylib#91] [Ideas]
Grabbing and using the headers to paginate the Rest API calls is really a pain. This is the kind of things we should be able to standardize and handle in SPyLib.

**Comment by qw-in** [comment]
Off the top of my head, something like this would be nice:

```python

class Token:
    def paginated_resource(path: str, **query_string) -> PaginatedResource
       ...
```

The `PaginatedResource` class then implements convenient methods such as an async iterator

Maybe it would be convenient if this shorthand worked as well?

```python

async for product in token.paginated_resource('/products'):
    #...
```
